### PR TITLE
Newtreewalker workspace

### PR DESCRIPTION
**Pull request by badex-ai**
##ISSUE
#226

DESCRIPTION:
Created a new treewalker workspace and added the dependencies but only commented out the changes in the previous file just in case you would like to make adjustments. I am not perfect with my typescript yet (still learning and would like to learn more) but if there are any other changes that I need to get this merged pls let me know. 

made changes to the
 i18n.taggedString function in "warning.ts" to include parenthesis (I don't know if this was a typo) 
added typescript interfaces "depwalker.ts" in the new folder.

### CODE REVIEW

**Comment by fraxken** [comment]
Hello @badex-ai,

- You need to add the new workspace to tsconfig.json file
- You also need a tsconfig in workspaces/treewalker (and remove the .gitkeep).
- Do you have a prettier plugin enabled? If yes think to disable it.
- Some files have a broken syntax and big spaces between codes (workspaces/treewalker/src/depWalker.ts) 
- You also need to migrate tests

And you need to update the original depWalker.js file to include the new dependency (all tests suites must run with no issues)

**Comment by PierreDemailly** [comment]
We also need a new ci workflow (`.github/workflows/treewalker.yml`)

File: workspaces/treewalker/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  "types": "./dist/index.d.ts",
+  "scripts": {
+    "build": "tsc",
+    "test-only": "glob -c \"tsx --test\" \"./test/**/*.spec.ts\"",
+    "test": "c8 -r html npm run test-only"
+  },
+  "files": [
+    "index.js",
+    "src"
+  ],
```
Comment (PierreDemailly) [inline workspaces/treewalker/package.json:16]:
```suggestion
  "files": [
    "dist"
  ],
```

File: workspaces/treewalker/package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "@nodesecure/treewalker",
```
Comment (PierreDemailly) [inline workspaces/treewalker/package.json:2]:
```suggestion
  "name": "@nodesecure/treeWalker",
```
or
```suggestion
  "name": "@nodesecure/tree-walker",
```

File: workspaces/treewalker/package.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "test-only": "glob -c \"tsx --test\" \"./test/**/*.spec.ts\"",
+    "test": "c8 -r html npm run test-only"
+  },
+  "files": [
+    "index.js",
+    "src"
+  ],
+  "keywords": [
+    "NodeSecure"
+  ],
```
Comment (PierreDemailly) [inline workspaces/treewalker/package.json:19]:
```suggestion
  "keywords": [
    "NodeSecure",
    "TreeWalker",
    "Tree",
    "Walker",
    "Dependencies",
    "Traversal"
  ],
```

File: workspaces/treewalker/package.json (hunk 2 above)
Comment (PierreDemailly) [reply workspaces/treewalker/package.json:2]: @fraxken Shouldn't even be `@nodesecure/depWalker` ?

File: workspaces/treewalker/package.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "@nodesecure/treewalker",
+  "version": "1.0.0",
+  "description": "NodeSecure treewalker",
```
Comment (PierreDemailly) [inline workspaces/treewalker/package.json:4]:
```suggestion
  "description": "Efficient dependencies walker",
```

**Comment by PierreDemailly** [comment]
Please also add a README

**Comment by fraxken** [comment]
Hello, I'm closing this in favor of https://github.com/NodeSecure/scanner/pull/236

Thanks for the effort / contribution.

**Pull request closed without merging**